lua/lspconfig/configs.lua

```diff
@@ -245,7 +245,13 @@ function configs.__newindex(t, config_name, config_def)
       end
 
       if root_dir then
-        id = manager.add(root_dir, false)
+        -- Check if there is an active server already for this same file type
+        local active_clients_list = util.get_active_clients_list_by_ft(vim.bo[bufnr].filetype)
+        if #active_clients_list ~= 0 then
+          id = active_clients_list[1].id
```

Comment:
I don't think there can use index 1 if there has multiple clients for this filetype . like efm tsserver for ts file the 1 can be efm or tsserver .

Reply:
We don't have the original lsp client id at this point, so we can't tell which client to choose

Reply:
I am trying to remove root_dir in lsp start client in core. maybe can solve this. there is no root_dir definition in lsp doc. use wrokspace folder would be better?

Reply:
I'm not sure if this is a correct approach , helix also have this concept , (emacs as well?)

I'm not sure what vscode does

Reply:
I asked a bit , seems like vscode sets the root workspace as root but it have more logic to add more workspaces (vscode can handle multiple workspaces at the time)

Honestly I can't answer here as I'm not familiar that much with lsp protocol

Reply:
you mean helix has root_dir ? emacs lsp-bridge, yegappan/lsp  don't have root_dir and the root_dir is not in lsp specification. it just used to check there already has an exist client and reused.

Reply:
That's what I understood when I asked in their matrix channel, seems they're still  designing this though https://github.com/helix-editor/helix/pull/4439

Reply:
I am not familiar with helix. But this doesn't look like a good way.

Reply:
maybe we can do it like this ?

```diff
--- a/lua/lspconfig/util.lua
+++ b/lua/lspconfig/util.lua
@@ -253,6 +253,14 @@ function M.server_per_root_dir_manager(make_config)
     -- Check if we have a client already or start and store it.
     if not client_id then
       local new_config = make_config(root_dir)
+      if next(clients) ~= nil then
+        for _, id in pairs(clients) do
+          local client = lsp.get_client_by_id(id)
+          if client.name == new_config.name then
+            return id
+          end
+        end
+      end
       -- do nothing if the client is not enabled
       if new_config.enabled == false then
         return
```

Reply:
I think that works  I updated the pr with your code

Reply:
do we need add the new root dir into the workspaceFolder ?

Reply:
I'm not sure when I tested with jonhoo original rust problem your snippet fixes it

Reply:
maybe them belong to one workspace.

Reply:
> maybe them belong to one workspace.

no, they belong different workspace when you try to open another project file